src/harness/sas_testcase.py

```diff
@@ -152,23 +152,36 @@ def TriggerDailyActivitiesImmediatelyAndWaitUntilComplete(self):
       time.sleep(10)
     signal.alarm(0)
     
-  def TriggerFullActivityDumpAndWaitUntilComplete(self):
+  def assertChannelsContainFrequencyRange(self, channels, frequency_range):
+    channels.sort(key=lambda ch: (ch['frequencyRange']['lowFrequency'], ch['frequencyRange']['highFrequency']), reverse = False)
+    for index, channel in  enumerate(channels):
+        if index == 0:
+           self.assertEqual(channel['frequencyRange']['lowFrequency'],\
+             frequency_range['lowFrequency'])
+        else:
+            self.assertLessEqual(channel['frequencyRange']['lowFrequency'],
+                                 channels[index - 1]\
+                                 ['frequencyRange']['highFrequency'])
+            
+    channels.sort(key=lambda ch: (ch['frequencyRange']['highFrequency']), reverse = True)
+    self.assertEqual(channels[0]['frequencyRange']['highFrequency'],\
+             frequency_range['highFrequency'])
+   
+  def TriggerFullActivityDumpAndWaitUntilComplete(self, server_cert, server_key):
     self._sas_admin.TriggerFullActivityDump()
     request_time = datetime.utcnow()
-    
     signal.signal(signal.SIGALRM,
                   lambda signum, frame:
                   (_ for _ in ()).throw(Exception('Full Activity Dump Check Timeout')))
-    dump_message = self._sas.GetFullActivityDump()
-    dump_time = datetime.strptime(dump_message['generationDateTime'],
-                                               '%Y-%m-%dT%H:%M:%SZ')
     # Timeout after 2 hours if it's not completed
     signal.alarm(7200)
-    # Check the Status of Daily Activities every 10 seconds
-    while request_time > dump_time:
-      time.sleep(10)
-      dump_message = self._sas.GetFullActivityDump()
-      dump_time = datetime.strptime(dump_message['generationDateTime'],
+    # Check generation date of full activity dump 
+    while True:
+	  dump_message = self._sas.GetFullActivityDump( server_cert, server_key)
```

Comment:
I think we need to allow the possibility of HTTP error from the SAS UUT. Behavior should be: if HTTP error, wait 10 seconds and try again (same as if the timestamp is too old).